public/inputServiceWorker.js

```diff
@@ -8,50 +8,56 @@ function sleep(ms) {
-        if(url.includes("input")){ // Special requests targeted at getting input from the user
-            if(event.request.method === "GET"){ // Request from the worker to receive input
+    if (url.includes(papyrosHost)) { // requests to our own domain
+        // Special requests targeted at getting input from the user
+        if (url.includes("__papyros_input")) {
+            if (event.request.method === "GET") { // Request from the worker to receive input
                 promiseChain = waitForInput();
-            } else if(event.request.method === "POST"){ // Request from Papyros to send input
+            } else if (event.request.method === "POST") { // Request from Papyros to send input
                 promiseChain = event.request.json().then(resp => {
                     workerData.input = resp.input;
                     return new Response("input stored: " + workerData.input);
-                })
+                });
             }
         } else { // Requests to general Papyros pages
             promiseChain = fetch(event.request)
-                .then((response) => {
-                // Add new headers to be able to use SharedArrayBuffers if the browser supports them
-                // See also https://stackoverflow.com/questions/64650119/react-error-sharedarraybuffer-is-not-defined-in-firefox
-                const newHeaders = new Headers(response.headers);
-                newHeaders.set("Cross-Origin-Embedder-Policy", "require-corp");
-                newHeaders.set("Cross-Origin-Opener-Policy", "same-origin");
-        
-                const moddedResponse = new Response(response.body, {
-                    status: response.status || 200,
-                    statusText: response.statusText,
-                    headers: newHeaders,
-                });
-        
-                return moddedResponse;
+                .then(response => {
+                    // Add new headers to be able to use SharedArrayBuffers if the browser supports them
+                    // See also https://stackoverflow.com/questions/64650119/react-error-sharedarraybuffer-is-not-defined-in-firefox
+                    const newHeaders = new Headers(response.headers);
+                    newHeaders.set("Cross-Origin-Embedder-Policy", "require-corp");
+                    newHeaders.set("Cross-Origin-Opener-Policy", "same-origin");
 
-                })
-            }
+                    const moddedResponse = new Response(response.body, {
+                        status: response.status || 200,
+                        statusText: response.statusText,
+                        headers: newHeaders,
+                    });
+                    return moddedResponse;
+                });
+        }
     } else { // requests to other domains are unaltered
         promiseChain = fetch(event.request);
     }
     event.respondWith(promiseChain);
 });
+// Prevent needing to reload page to have working input
```

Comment:
Does this mean the cross origin isolation headers will sometimes be provided by the server instead of the service worker?

Reply:
This is to allow the input url to work, in the case where there are no SharedArrayBuffers. Ideally, the server always provides the headers because a reload is required if the headers are not yet there upon page load (as the service worker was not active yet and could not add them), but I cannot do this without harming the user experience.